src/components/Search.tsx

```diff
@@ -1,68 +1,78 @@
-import type {GestureResponderEvent, StyleProp, ViewStyle} from 'react-native';
-import {View} from 'react-native';
-import useLocalize from '@hooks/useLocalize';
-import useTheme from '@hooks/useTheme';
-import useThemeStyles from '@hooks/useThemeStyles';
-import variables from '@styles/variables';
-import CONST from '@src/CONST';
-import Icon from './Icon';
-import * as Expensicons from './Icon/Expensicons';
-import {PressableWithFeedback} from './Pressable';
-import Text from './Text';
-import Tooltip from './Tooltip';
+import React, {useEffect} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import useNetwork from '@hooks/useNetwork';
+import * as SearchActions from '@libs/actions/Search';
+import * as SearchUtils from '@libs/SearchUtils';
+import EmptySearchView from '@pages/Search/EmptySearchView';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import isLoadingOnyxValue from '@src/types/utils/isLoadingOnyxValue';
+import TableListItemSkeleton from './Skeletons/TableListItemSkeleton';
+
+// For testing purposes run this code in browser console to insert fake data:
+// query is the param from URL, by default it will be "all"
+// Onyx.set(`search_${query}`, {
+//             search: {
+//                 offset: 0,
+//                 type: 'transaction',
+//                 hasMoreResults: false,
+//             },
+//             data: {
+//                 transactions_1234: {
+//                     receipt: {source: 'http...'},
+//                     hasEReceipt: false,
+//                     created: '2024-04-11 00:00:00',
+//                     amount: 12500,
+//                     type: 'cash',
+//                     reportID: '1',
+//                     transactionThreadReportID: '2',
+//                     transactionID: '1234',
+//                 },
+//                 transactions_5555: {
+//                     receipt: {source: 'http...'},
+//                     hasEReceipt: false,
+//                     created: '2024-04-11 00:00:00',
+//                     amount: 12500,
+//                     type: 'cash', // not present in live data (data outside of snapshot_)
+//                     reportID: '1',
+//                     transactionThreadReportID: '2',
+//                     transactionID: '5555',
+//                 },
+//             },
+//         })
 
 type SearchProps = {
-    // Callback fired when component is pressed
-    onPress: (event?: GestureResponderEvent | KeyboardEvent) => void;
+    query: string;
+};
 
-    // Text explaining what the user can search for
-    placeholder?: string;
+function Search({query}: SearchProps) {
+    const {isOffline} = useNetwork();
+    const [searchResults, searchResultsMeta] = useOnyx(`${ONYXKEYS.COLLECTION.SEARCH}${query}`);
 
-    // Text showing up in a tooltip when component is hovered
-    tooltip?: string;
+    useEffect(() => {
+        SearchActions.search(query);
+    }, [query]);
```

Comment:
Hmm I found it to be a bit weird that refreshing the page doesn't trigger a call to search, but that's something we could iron out later in a follow up if needed.

Reply:
I believe the `search()` gets called, but its the implementation of `isNetworkOffline` thats breaking it.
I did some logging and in the beginning the NETWORK sets `isOffline` to true for a split second, and that is when the useEffect triggers. Then a bit later `isOffline` is correctly set to `false` but by then effect has already run and no search call was made.

Not sure how to fix this, need some guidance from you. Is that the intended behavior of `ONYXKEYS.NETWORK`?

Reply:
I don't think that's a blocker for now. Let's move on and address this as a follow up.